chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -254,8 +254,12 @@ def self.all(node)
           node[:block_device].keys.map{|d|Disk.new(node,d)}
         end
 
-        def self.unclaimed(node)
+        def self.unclaimed(node,mounted=false)
           all(node).select do |d|
+            unless mounted
+              %x{lsblk #{d.name} --noheadings --output MOUNTPOINT | grep -q -v ^$}
```

Comment:
I think you need to keep the `gsub(/!/, "/")` for cciss disks.

I'm also unsure about the grep command, but that's because my brain is slow right now :/ It will exit with 0 if there are non-empty lines, and with non-0 if there are non-empty lines, is that correct?